src/db/spanner/models.rs

```diff
@@ -63,6 +67,9 @@ struct SpannerDbSession {
     /// Currently locked collections
     coll_locks: HashMap<(HawkIdentifier, i32), CollectionLock>,
     transaction: Option<TransactionSelector>,
+    /// Behind Vec so commit can take() it (maybe commit() should consume self
+    /// instead?)
```

Comment:
I kind of agree that `commit()` consuming self is more in spirit with how I understand ownership should work. You shouldn't be able to do anything else with the session once the commit has happened.